### ISSUE TITLE
Suppliers error when adding new combination - PrestaShop\PrestaShop\Core\Localization\Exception\LocalizationException kod 0

### ISSUE THREAD

**Issue by Prestaworks** [PrestaShop/PrestaShop#22782]
#### Describe the bug

Error [PrestaShop\PrestaShop\Core\Localization\Exception\LocalizationException kod 0] is shown when clicking on the magnifier for a supplier.

#### Expected behavior

List of products connected to the supplier.

#### Steps to Reproduce

1. Go to suppliers
2. create a supplier
3. Go to a product id 2 in the default data (Hummingbird printed sweater), connect this to the supplier and enter prices for purchase for all sizes.
4. Go to supplier and click on the magnifier for the supplier, product is displayed.
5. Go to attributes, click on size.
6. Add a new size XXXL (do NOT click on the save button for the product)
7. Go back to product id 2 and add this new size to the product.
8. Go to supplier and click on the magnifier for the supplier, you now get redirected to supplier list and the error is displayed.

#### Additional information

* PrestaShop version: 1.7.7.0
* PHP version: 7.3

Issue is in
`\src\Adapter\Supplier\QueryHandler\GetSupplierForViewingHandler.php`
```
$this->locale->formatPrice(
                                $productInfo['product_supplier_price_te'],
                                Currency::getIsoCodeById((int) $productInfo['id_currency'])
                            );
```
The productInfo will be "null" here so formatPrice breaks because the iso_code is "".

**Comment by hibatallahAouadni** [comment]
Hello @Prestaworks 

I encounter another issue in the fourth step:
> Go to supplier and click on the magnifier for the supplier, product is displayed.

In the product page, I add 5 euro for Size S:

![image](https://user-images.githubusercontent.com/23308427/104351040-26422a80-5505-11eb-86da-e25028a51a88.png)

In the Supplier page, I get this:

![image](https://user-images.githubusercontent.com/23308427/104351079-335f1980-5505-11eb-9739-f59ad8724df9.png)

Did you encounter this issue too? as you can see all the prices are wrongly displayed :confused: 
Please check and feedback.

Thanks!

**Comment by Prestaworks** [comment]
Hi @hibatallahAouadni 
I tried to recreate your issue with the price, but it worked as it should for me. I even tried adding extra currencies and changing the price and currency, but it was always working correctly for me.

**Comment by hibatallahAouadni** [comment]
That's weird :confused: 
Ping @khouloudbelguith could please reproduce my issue :wink: Thanks in advance :slightly_smiling_face:

**Comment by khouloudbelguith** [comment]
Hi @Prestaworks, @hibatallahAouadni,

Yes, I manage to reproduce the issue with PS1771.
![image](https://user-images.githubusercontent.com/16067358/104484535-afbb3080-55c9-11eb-87b1-81a05f8cb377.png)

I attached a screen record
https://drive.google.com/file/d/1kjwWPT2nZmqNed6shVdFFoU3Cu14zE89/view?usp=sharing

Thank you!

**Comment by hibatallahAouadni** [comment]
Thanks for help @khouloudbelguith :wink: 
So, I’ll add this to the debug backlog so that it’s fixed.
If you have already fixed it on your end or if you think you can do it, please do send us a pull request!

Thanks!

**Comment by florine2623** [comment]
This issue is fixed on PS 9.0.0

I am redirected to the list of products for the supplier :
![Image](https://github.com/user-attachments/assets/12b39d98-cbc7-4e36-8ff7-a5597c0994d5)
